src/server/lib/encryptedStateCookie.ts

```diff
@@ -61,7 +62,20 @@ export const readEncryptedStateCookie = (
         encryptedCookie,
         getConfiguration().encryptionSecretKey, // prevent the key from lingering in memory by only calling when needed
       );
-      return JSON.parse(decrypted);
+      const decryptedState = JSON.parse(decrypted);
+
+      // validate 'boolean' string values
+      return {
+        email: decryptedState.email,
+        emailType: decryptedState.emailType,
+        passwordSetOnWelcomePage: isStringBoolean(
+          decryptedState.passwordSetOnWelcomePage,
+        ),
+        status: decryptedState.status,
+        signInRedirect: isStringBoolean(decryptedState.signInRedirect),
+        cmpConsentedState: isStringBoolean(decryptedState.cmpConsentedState),
+        queryParams: decryptedState.queryParams,
+      };
```

Comment:
you don't actually need to validate the string boolean values here, the data was encoded and encrypted as JSON which supports boolean values!

Reply:
so you can leave it as 
```suggestion
return JSON.parse(decrypted);
```